### PR TITLE
opencode: add themes option

### PR DESCRIPTION
**Pull request by arunoruto**
Add themes option to configure custom themes through nix. Define a submodule for the settings option, to give the `theme` setting a description.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `nix fmt` or
      `nix-shell -p treefmt nixfmt deadnix keep-sorted --run treefmt`.

- [x] Code tested through `nix run .#tests -- test-all` or
      `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

  ```
  {component}: {description}

  {long description}
  ```

  See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/a51598236f23c89e59ee77eb8e0614358b0e896c/modules/programs/lesspipe.nix#L11).
  - [ ] Generate a news entry. See [News](https://nix-community.github.io/home-manager/index.xhtml#sec-news)
  - [ ] Basic tests added. See [Tests](https://nix-community.github.io/home-manager/index.xhtml#sec-tests)

- If this PR adds an exciting new feature or contains a breaking change.
  - [ ] Generate a news entry. See [News](https://nix-community.github.io/home-manager/index.xhtml#sec-news)

@delafthi

### CODE REVIEW

File: modules/programs/opencode.nix [hunk 1]
```diff
@@ -26,7 +26,15 @@ in
...
-      inherit (jsonFormat) type;
+      type = lib.types.submodule {
+        freeformType = jsonFormat.type;
+        options.theme = mkOption {
+          type = lib.types.str;
+          default = "opencode";
+          example = "catppuccin";
+          description = "The theme to use for opencode.";
+        };
+      };
```
Comment (delafthi) [inline modules/programs/opencode.nix]: I don't see the advantage of that. Adding this submodule makes the `empty-settings` test fail, because the settings are never empty (though I might be wrong about that).

File: modules/programs/opencode.nix [hunk 2]
```diff
@@ -137,6 +145,15 @@ in
...
     };
 
+    themes = mkOption {
+      inherit (jsonFormat) type;
+      default = { };
+      description = ''
+        Custom themes for opencode. The attribute name becomes the command filename.
+        Commands are stored in $XDG_CONFIG_HOME/opencode/themes/ directory.
+        See <https://opencode.ai/docs/themes/> for the documentation.
+      '';
```
Comment (delafthi) [inline modules/programs/opencode.nix]:
```suggestion
        Custom themes for opencode. The attribute name becomes the theme filename.
        Themes are stored in {file}`$XDG_CONFIG_HOME/opencode/themes/` directory.
        See <https://opencode.ai/docs/themes/> for the documentation.
      '';
```

File: modules/programs/opencode.nix [hunk 3]
```diff
@@ -137,6 +145,15 @@ in
       '';
     };
 
+    themes = mkOption {
+      inherit (jsonFormat) type;
```
Comment (delafthi) [inline modules/programs/opencode.nix]: I would also prefer if one could provide either text or a path to a file.


```suggestion
      type = lib.types.attrsOf (lib.types.either jsonFormat lib.types.path);
```

File: tests/modules/programs/opencode/default.nix [hunk 4]
```diff
@@ -8,4 +8,5 @@
   opencode-agents-path = ./agents-path.nix;
   opencode-commands-path = ./commands-path.nix;
   opencode-mixed-content = ./mixed-content.nix;
+  opencode-themes = ./themes.nix;
```
Comment (delafthi) [inline tests/modules/programs/opencode/default.nix:11]: Please also add an 'empty-themes' check to verify that no files are created when the `themes` is empty.

File: tests/modules/programs/opencode/my-theme.nix
Comment (delafthi) [inline tests/modules/programs/opencode/my-theme.nix:1]: I would merge this file into `tests/modules/programs/opencode/themes.nix`.

File: tests/modules/programs/opencode/my-theme.nix [hunk 5]
```diff
@@ -0,0 +1,223 @@
+{
+  "$schema" = "https://opencode.ai/theme.json";
```
Comment (delafthi) [inline tests/modules/programs/opencode/my-theme.nix]: Please add the schema dynamically to the configuration. See https://github.com/nix-community/home-manager/blob/722792af097dff5790f1a66d271a47759f477755/modules/programs/opencode.nix#L148-L151

File: modules/programs/opencode.nix (hunk 1 above)
Comment (arunoruto) [reply modules/programs/opencode.nix]: I thought that since the themes option is introduced, it would be good to document the setting option that enables a respective theme.
But if `empty-settings` test is failing, and it can't be empty anymore, maybe I can just document the procedure in the description of the themes option.

**Comment by arunoruto** [comment]
> Hey, thanks for your contribution - great addition to the opencode home module.
> 
> Make sure that all tests pass (see [manual-test-commands](https://github.com/nix-community/home-manager/blob/master/docs/manual/contributing/tests.md#manual-test-commands-sec-tests-manual)).

Thanks for the tips!
```log
nix run .#tests -- opencode                   on  HEAD (dd3468d) (REBASING 1/1) [✘!+]
ℹ️ Discovering tests...
ℹ️ Running 10 test(s)...

--- Running test 1/10: test-opencode-agents-inline ---
✅ Test passed: test-opencode-agents-inline

--- Running test 2/10: test-opencode-agents-path ---
✅ Test passed: test-opencode-agents-path

--- Running test 3/10: test-opencode-commands-inline ---
✅ Test passed: test-opencode-commands-inline

--- Running test 4/10: test-opencode-commands-path ---
✅ Test passed: test-opencode-commands-path

--- Running test 5/10: test-opencode-empty-rules ---
✅ Test passed: test-opencode-empty-rules

--- Running test 6/10: test-opencode-empty-settings ---
✅ Test passed: test-opencode-empty-settings

--- Running test 7/10: test-opencode-mixed-content ---
✅ Test passed: test-opencode-mixed-content

--- Running test 8/10: test-opencode-rules ---
✅ Test passed: test-opencode-rules

--- Running test 9/10: test-opencode-settings ---
✅ Test passed: test-opencode-settings

--- Running test 10/10: test-opencode-themes ---
✅ Test passed: test-opencode-themes

--- Summary ---
✅ All 10 tests passed!
```

File: tests/modules/programs/opencode/my-theme.nix
Comment (arunoruto) [reply tests/modules/programs/opencode/my-theme.nix:1]: I put the `nmt` section above, so the one doesn't have to scroll to the bottom to see the tests. I hope this is okay.

**Review by delafthi — APPROVED** [review]
